### PR TITLE
refactor: split config update based on component

### PR DESCRIPTION
**Pull request by osmman**

### CODE REVIEW

**Comment by osmman** [comment]
I am splitting change into multiple PRs

File: test/e2e/common_install_test.go [hunk 1]
```diff
@@ -84,7 +90,7 @@ var _ = Describe("Securesign install with certificate generation", Ordered, func
 					},
 				},
 				Trillian: v1alpha1.TrillianSpec{Db: v1alpha1.TrillianDB{
-					Create: utils.Pointer(true),
+					Create: ptr.To(true),
```
Comment (lance) [inline test/e2e/common_install_test.go:93]: Just curious what the difference between `ptr.To()` and `utils.Pointer()` is.

File: test/e2e/common_install_test.go [hunk 2]
```diff
@@ -7,15 +7,21 @@ import (
...
 
+	"github.com/securesign/operator/test/e2e/support/tas/tsa"
+
+	"k8s.io/utils/ptr"
+
+	"github.com/securesign/operator/test/e2e/support/tas"
+
 	"github.com/securesign/operator/internal/controller/common/utils"
+	"github.com/securesign/operator/test/e2e/support/tas/fulcio"
+	"github.com/securesign/operator/test/e2e/support/tas/rekor"
```
Comment (lance) [inline test/e2e/common_install_test.go:18]: I like this replacement of `"github.com/securesign/operator/test/e2e/support/tas"` with the individual component packages. That seems more maintainable in the long run.

File: test/e2e/common_install_test.go (hunk 1 above)
Comment (osmman) [reply test/e2e/common_install_test.go:93]: Implementation should be same. The reason is to migrate to k8s utils implementation and remove utils.Pointer from project. There has been problems with cyclical imports for some unit tests.

File: test/e2e/upgrade_test.go [hunk 3]
```diff
@@ -10,6 +10,16 @@ import (
 	"strings"
 	"time"
 
+	"github.com/securesign/operator/test/e2e/support/tas/ctlog"
+
+	"github.com/securesign/operator/test/e2e/support/tas/tsa"
+
+	fulcio2 "github.com/securesign/operator/test/e2e/support/tas/fulcio"
```
Comment (bouskaJ) [inline test/e2e/upgrade_test.go:17]: I believe we can choose better package naming.

File: test/e2e/upgrade_test.go [hunk 4]
```diff
@@ -203,7 +210,7 @@ var _ = Describe("Operator upgrade", Ordered, func() {
 					Certificate: tasv1alpha.FulcioCert{
 						OrganizationName:  "MyOrg",
 						OrganizationEmail: "my@email.org",
-						CommonName:        "fulcio",
+						CommonName:        "fulcioAction",
```
Comment (bouskaJ) [inline test/e2e/upgrade_test.go:213]:
```suggestion
						CommonName:        "fulcio",
```

Comment (osmman) [reply test/e2e/upgrade_test.go:213]: thank to catching it, it looks like that refactor rename in IDE function changed more things that should. I will fix that

Comment (osmman) [reply test/e2e/upgrade_test.go]: fixed

**Pull request merged**